llvm/lib/Support/TimeProfiler.cpp

```diff
@@ -104,6 +105,23 @@ struct llvm::TimeTraceProfilerEntry {
   }
 };
 
+struct InProgressEntry {
+  std::unique_ptr<TimeTraceProfilerEntry> Event;
+  std::vector<TimeTraceProfilerEntry> InstantEvents;
```

Comment:
Can you add documentation about InstantEvents and that they are associated with a parent duration and they are dropped if parent duration is shorter than the granularity.